marketplace/commercetools-sku/src/index.js

```diff
@@ -0,0 +1,173 @@
+
+  const ID = 'dialog-root';
+  const container = document.createElement('div');
+  container.id = ID;
+  container.style.display = 'flex';
+  container.style.flexDirection = 'column';
+  document.body.appendChild(container);
+
+  renderSkuPicker(ID, {
+    sdk,
+    fetchProductPreviews,
+    fetchProducts: memoizePromise(async (search, pagination) => {
+      const PER_PAGE = 20;
+      const requestBuilder = createRequestBuilder({ projectKey });
+      const uri = requestBuilder.productProjectionsSearch
+        .parse({
+          ...(!!search.length && {
+            text: {
+              language: locale,
+              value: search
+            }
+          }),
+          page: pagination.offset / PER_PAGE + 1,
+          perPage: PER_PAGE,
+          sort: [{ by: `name.${locale}`, direction: 'asc' }]
+        })
+        .build();
+      const response = await client.execute({ uri, method: 'GET' });
+      if (response.statusCode === 200) {
+        return {
+          pagination: {
+            count: response.body.count,
+            limit: response.body.limit,
+            total: response.body.total,
+            offset: response.body.offset
+          },
+          products: response.body.results.map(dataTransformer(sdk.parameters.installation))
+        };
+      }
+      throw new Error(response.statusCode);
+    })
+  });
+
+  sdk.window.updateHeight(window.outerHeight);
+}
+
+async function openDialog(sdk, currentValue, config) {
+  const maxFiles = config.maxFiles - currentValue.length;
+
+  const skus = await sdk.dialogs.openExtension({
+    position: 'center',
+    title: makeCTA(sdk.field.type),
+    shouldCloseOnOverlayClick: true,
+    shouldCloseOnEscapePress: true,
+    parameters: { ...config, maxFiles },
+    width: 1400
+  });
+
+  return Array.isArray(skus) ? skus : [];
+}
+
+function isDisabled(/* currentValue, config */) {
+  // No restrictions need to be imposed as to when the field is disabled from the app's side
+  return false;
+}
+
+function validateParameters(parameters) {
+  if (parameters.projectKey.length < 1) {
+    return 'Provide your Commercetools project key.';
+  }
+
+  if (parameters.clientId.length < 1) {
+    return 'Provide your Commercetools client ID.';
+  }
+
+  if (parameters.clientSecret.length < 1) {
+    return 'Provide your Commercetools client secret.';
+  }
+
+  if (parameters.apiEndpoint.length < 1) {
+    return 'Provide the Commercetools API endpoint.';
+  }
+
+  if (parameters.authApiEndpoint.length < 1) {
+    return 'Provide the Commercetools auth API endpoint.';
+  }
+
+  if (parameters.locale.length < 1) {
+    return 'Provide the Commercetools data locale.';
+  }
+
+  return null;
+}
+
+setup({
+  makeCTA,
+  name: 'Commercetools',
+  logo,
+  description:
+    'The Commercetools app is a widget that allows editors to select products from their Commercetools account. Select a product from Commercetools that you want your entry to reference.',
```

Comment:
This description doesn't really make sense when you read it... I would recommend this instead.

```suggestion
    'The Commercetools app allows editors to select products from their Commercetools account and reference them inside of Contentful entries.
```